src/apis/openai.test.ts

```diff
@@ -56,19 +56,24 @@ test("gets chat completion (stream)", async () => {
     throw new Error("completion is not readable stream");
   }
 
-  const completion = [];
+  let message = "";
+  let firstChunk: any;
+  let lastChunk: any;
   const reader = stream.getReader();
   while (true) {
     const { done, value } = await reader.read();
     if (done) {
       break;
     }
-    completion.push(JSON.parse(new TextDecoder().decode(value)));
+    const chunk = JSON.parse(new TextDecoder().decode(value));
+    message += chunk?.choices[0]?.delta?.content || "";
+    if (!firstChunk) {
+      firstChunk = chunk;
+    }
+    lastChunk = chunk;
```

Comment:
The test passed, but when I tried out with a real OpenAI API client library, it does not work. I also tested with Open WebUI, it does not work.

Therefore, to increase the confidence, I went ahead and installed the official OpenAI SDK for use inside the test. This will ensure that the mock endpoint is compliant with OpenAI’s request format.